src/views/RequestLinkForm/index.tsx

```diff
@@ -0,0 +1,435 @@
+
+  const [setSystemIntakeRelationExistingSystem] = useMutation<
+    SetSystemIntakeRelationExistingSystem,
+    SetSystemIntakeRelationExistingSystemVariables
+  >(SetSystemIntakeRelationExistingSystemQuery);
+
+  const [setSystemIntakeRelationExistingService] = useMutation<
+    SetSystemIntakeRelationExistingService,
+    SetSystemIntakeRelationExistingServiceVariables
+  >(SetSystemIntakeRelationExistingServiceQuery);
+
+  const [relation, setRelation] = useState<RequestRelation | null>(null);
+
+  const taskListUrl = `/governance-task-list/${systemId}`;
+
+  const { control, watch, handleSubmit } = useForm<{
+    cedarSystemIDs: string[];
+    contractNumbers: string;
+    contractName: string;
+  }>({
+    defaultValues: {
+      cedarSystemIDs: [],
+      contractNumbers: '',
+      contractName: ''
+    }
+  });
+
+  // Ref fields for some form behavior
+  const fields = watch();
+
+  // This form doesn't use validation
+  // The submission button is disabled according to required fields
+  const submitEnabled = (() => {
+    if (relation === null) return false;
+
+    if (relation === 'newSystem') return true;
+
+    if (relation === 'existingSystem' && fields.cedarSystemIDs.length)
+      return true;
+
+    if (relation === 'existingService' && fields.contractName.trim() !== '')
+      return true;
+
+    // Default to disabled
+    return false;
+  })();
+
+  const submit = handleSubmit(data => {
+    // The new system relation form is entirely optional
+    // If it's empty then just treat this submit handler as a link
+    if (relation === 'newSystem' && data.contractNumbers.trim() === '') {
+      history.push(taskListUrl);
+      return;
+    }
+
+    // Otherwise do some field parsing and correlate `relation` to mutation
+
+    // Parse contract numbers from csv text input to string[]
+    // Make sure an empty string input is sent as an empty list
+    const contractNumbers = data.contractNumbers
+      .split(',')
+      .map(v => v.trim())
+      .filter(v => v !== '');
+
+    let p: Promise<any> | undefined;
+
+    if (relation === 'newSystem') {
+      p = setSystemIntakeRelationNewSystem({
+        variables: {
+          input: {
+            systemIntakeID: systemId,
+            contractNumbers
+          }
+        }
+      });
+    } else if (relation === 'existingSystem') {
+      p = setSystemIntakeRelationExistingSystem({
+        variables: {
+          input: {
+            systemIntakeID: systemId,
+            cedarSystemIDs: data.cedarSystemIDs,
+            contractNumbers
+          }
+        }
+      });
+    } else if (relation === 'existingService') {
+      p = setSystemIntakeRelationExistingService({
+        variables: {
+          input: {
+            systemIntakeID: systemId,
+            contractName: data.contractName,
+            contractNumbers
+          }
+        }
+      });
+    }
+
+    p?.then(() => {
+      history.push(taskListUrl);
+    });
```

Comment:
I think we should have some error handling here.  I don't see any nocked in Figma, but I think FE should always handle this possibility with some generic alert

Reply:
OK, I'll find something to put there. I let it slide since it seemed like one of those cases where if something wrong happened in that submit, then there are probably bigger problems with the app. Maybe some feedback to retry would be a little nicer.